core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -55,37 +59,56 @@ void shouldCheckIfTheUnreleasedCommitsExist() throws IOException, GitAPIExceptio
 //        ref: https://www.vogella.com/tutorials/JGit/article.html
 //        creating a fake Git repository
 
-        Path parentDir = Paths.get("/path/to/local");
-        Path tempDir = Files.createTempDirectory(parentDir, "mytempdir");
-
-        Path srcTempDir = Files.createTempDirectory(tempDir, "filedir");
-//        create a commit
+        Path parentDir = Paths.get("/path/to/repository");
+        //        create a commit
 //        https://stackoverflow.com/a/51151158
-        Git git = Git.init().setDirectory(tempDir.toFile()).call();
+        Git git = Git.init().setDirectory(parentDir.toFile()).call();
 
         // create files to add to the repo
 
-        File file1 = new File(tempDir.toString(), "pom.xml");
-        File file2 = new File(tempDir.toString(), "README.md");
-        File file3 = new File(tempDir.toString(), String.valueOf(srcTempDir));
+        File file1 = new File(parentDir.toString(), "pom.xml");
+        File file2 = new File(parentDir.toString(), "README.md");
+        File directory = new File(parentDir+File.separator+"src"+ File.separator+"main"+File.separator+"resources"+File.separator+"test.txt");
+
         file1.createNewFile();
         file2.createNewFile();
-        file3.createNewFile();
+
 
         AddCommand add = git.add();
         // Stage all files in the repo
-        add.addFilepattern("pom.xml").addFilepattern("README.md").addFilepattern("src/test/resources/file.txt").call();
-        Repository repository = new FileRepositoryBuilder().setGitDir(tempDir.toFile()).build();
+        add.addFilepattern(file1.getPath());
+        add.addFilepattern(file2.getPath());
+        add.addFilepattern(directory.getPath());
+        add.call();
+
+        Repository repository = new FileRepositoryBuilder().setGitDir(parentDir.toFile()).build();
 
         // create committer
         PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
         PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().plusDays(1).toInstant()));
 
-        // commit the files
-        CommitCommand commit = git.commit();
-        commit.setOnly( "pom.xml" ).setOnly( "README.md" ).setOnly("src/test/resources/file.txt").
-            setCommitter(committer).
-            setMessage("initial commit").call();
+
+        Path filePath = Paths.get("/path/to/repository/pom.xml");
+        if (Files.exists(filePath)) {
+            System.out.println("pom.xml file exists!");
+        } else {
+            System.out.println("pom.xml file does not exist.");
+        }
+
+        // commit file1
+        CommitCommand commit1 = git.commit();
+        commit1.setOnly(file1.getPath()).setCommitter(committer).
+            setMessage("added pom.xml").call();
```

Comment:
This line throws an error:
```
org.eclipse.jgit.api.errors.JGitInternalException: Entry not found by path: \path\to\repository\pom.xml
```

I used debugger and checked the path. The file exists. It is also being added in the `addFilePattern()` the path is same. 

I need help to resolve this.